apps/home/app/layout.tsx

```diff
@@ -1,10 +1,37 @@
+'use client'
+
 import '@hamsurang/ui/globals.css'
+import { isValidEventOrigin } from '@hamsurang/utils'
 import { Inter } from 'next/font/google'
-import { type PropsWithChildren, Suspense } from 'react'
+import { useRouter } from 'next/navigation'
+import { type PropsWithChildren, Suspense, useEffect } from 'react'
 
 const inter = Inter({ subsets: ['latin'] })
 
 export default function RootLayout({ children }: PropsWithChildren): JSX.Element {
+  const router = useRouter()
+
+  useEffect(() => {
+    const handleIncomingMessage = (event: MessageEvent) => {
+      if (!isValidEventOrigin(event.origin)) {
+        return
+      }
+
+      const { data } = event
```

Comment:
```suggestion
    const handleIncomingMessage = (({ data, origin }): MessageEvent) => {
      if (!isValidEventOrigin(origin)) {
        return
      }
```

이건 어떠신가용?!
어차피 구조분해할당 할거라면, 콜백함수에서 해버리는 게 낫지않을까? 라는 생각에 남겨봅니다!

Reply:
[959c30c](https://github.com/hamsurang/origin/pull/37/commits/959c30c568fedac7af431ece05475037b06c6e4b)에서 반영 완료!